app/services/foreman/deprecation.rb

```diff
@@ -0,0 +1,13 @@
+module Foreman
+  class Deprecation
+    #deadline_version - is the version the deprecation is going to be deleted, the format must be a major release e.g "1.8"
+    def self.deprecation_warning(foreman_version_deadline, info)
+      raise Foreman::Exception.new(N_("Invalid version format, please enter in x.y (only major version).")) unless foreman_version_deadline.to_s.match(/\A\d[.]\d+\z/)
+      ActiveSupport::Deprecation.warn("You are using a deprecated behavior, it will be removed in version #{foreman_version_deadline}, #{info}", caller)
```

Comment:
No, Host::Managed doesn't have a find_by_ip implemented in it so it activates the method from Host:Base, that's why you are seeing this as the source of the deprecation.